lib/saluki-env/src/autodiscovery/mod.rs

```diff
@@ -0,0 +1,348 @@
+    /// Service ID
+    pub service_id: MetaString,
+    /// Tagger entity
+    pub tagger_entity: MetaString,
+    /// Whether this is a cluster check
+    pub cluster_check: bool,
+    /// Node name
+    pub node_name: MetaString,
+    /// Source of the configuration
+    pub source: MetaString,
+    /// Whether to ignore autodiscovery tags
+    pub ignore_autodiscovery_tags: bool,
+    /// Whether metrics are excluded
+    pub metrics_excluded: bool,
+    /// Whether logs are excluded
+    pub logs_excluded: bool,
+}
+
+impl From<ProtoConfig> for Config {
+    fn from(proto: ProtoConfig) -> Self {
+        // Convert advanced AD identifiers from proto
+        let advanced_ad_identifiers = proto
+            .advanced_ad_identifiers
+            .into_iter()
+            .map(|adv_id| {
+                let kube_service = adv_id.kube_service.map(|svc| KubeNamespacedName {
+                    name: svc.name.into(),
+                    namespace: svc.namespace.into(),
+                });
+
+                let kube_endpoints = adv_id.kube_endpoints.map(|endpoints| KubeNamespacedName {
+                    name: endpoints.name.into(),
+                    namespace: endpoints.namespace.into(),
+                });
+
+                AdvancedADIdentifier {
+                    kube_service,
+                    kube_endpoints,
+                }
+            })
+            .collect();
+
+        let init_config = bytes_to_hasmap(proto.init_config).unwrap_or_default();
+        let instances = proto
+            .instances
+            .into_iter()
+            .map(|instance| bytes_to_hasmap(instance).unwrap_or_default())
+            .collect();
+
+        Self {
+            name: proto.name.into(),
+            init_config,
+            instances,
+            metric_config: bytes_to_hasmap(proto.metric_config).unwrap_or_default(),
+            logs_config: bytes_to_hasmap(proto.logs_config).unwrap_or_default(),
+            ad_identifiers: proto.ad_identifiers.into_iter().map(MetaString::from).collect(),
+            advanced_ad_identifiers,
+            provider: proto.provider.into(),
+            service_id: proto.service_id.into(),
+            tagger_entity: proto.tagger_entity.into(),
+            cluster_check: proto.cluster_check,
+            node_name: proto.node_name.into(),
+            source: proto.source.into(),
+            ignore_autodiscovery_tags: proto.ignore_autodiscovery_tags,
+            metrics_excluded: proto.metrics_excluded,
+            logs_excluded: proto.logs_excluded,
+        }
+    }
+}
+
+fn bytes_to_hasmap(bytes: Vec<u8>) -> Result<HashMap<MetaString, serde_yaml::Value>, GenericError> {
+    let parse_bytes = String::from_utf8(bytes)?;
+
+    let map: HashMap<String, serde_yaml::Value> = serde_yaml::from_str(&parse_bytes)?;
+
+    let mut result = HashMap::<MetaString, serde_yaml::Value>::new();
+
+    for (key, value) in map {
+        result.insert(key.into(), value);
+    }
+
+    Ok(result)
+}
```

Comment:
For the sake of this method, we can just have it spit out `HashMap<String, serde_yaml::Value>` to avoid to complete rebuilding of the map just to convert the `String` to `MetaString`.